### PR TITLE
Refactor web driver

### PR DESCRIPTION
**Pull request by lamaks**
#### Description of changes

Refactor web driver and page scanner to be in sync with service implementation and address false positive color-contrast issue.

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)

### CODE REVIEW

File: packages/cli/src/web-driver/web-driver.ts [hunk 1]
```diff
@@ -1,27 +1,23 @@
...
 
     constructor(private readonly puppeteer: typeof Puppeteer = Puppeteer) {}
 
     public async launch(chromePath?: string): Promise<Puppeteer.Browser> {
-        let launchOption: Puppeteer.LaunchOptions;
-        launchOption = {
+        this.browser = await this.puppeteer.launch({
             headless: true,
-            timeout: 30000,
+            executablePath: chromePath,
```
Comment (AhmedAbdoOrtiga) [inline packages/cli/src/web-driver/web-driver.ts:16]: if chromePath is null, will puppeteer pick up the default or set it to null?

File: packages/cli/src/web-driver/web-driver.spec.ts [hunk 2]
```diff
@@ -60,4 +63,11 @@ describe('WebDriver', () => {
         expect(browser).toEqual(puppeteerBrowserMock);
         puppeteerLaunchMock.verifyAll();
     });
+
+    it('should update user agent string', async () => {
+        await testSubject.launch();
+
+        expect(testSubject.userAgent).toEqual('Chrome user agent string');
+        puppeteerLaunchMock.verifyAll();
```
Comment (AhmedAbdoOrtiga) [inline packages/cli/src/web-driver/web-driver.spec.ts:71]: can we move this to afterEach

File: packages/cli/src/scanner/page.ts [hunk 3]
```diff
@@ -100,17 +112,45 @@ export class Page {
...
         }
 
         return scanResults;
     }
 
-    public async close(): Promise<void> {
-        if (this.puppeteerPage !== undefined) {
-            await this.puppeteerPage.close();
-            await this.webDriver.close();
+    private async waitForPageToCompleteRendering(page: Puppeteer.Page, timeoutMsecs: number): Promise<void> {
```
Comment (AhmedAbdoOrtiga) [inline packages/cli/src/scanner/page.ts:123]: Will we have this issue in the crawler?

File: packages/cli/src/scanner/page.ts [hunk 4]
```diff
@@ -39,25 +43,29 @@ export class Page {
...
-        const gotoUrlPromise = this.puppeteerPage.goto(url, { waitUntil: ['load'], timeout: 120000 });
-        const networkLoadTimeoutInMilleSec = 30000;
-        const waitForNetworkLoadPromise = this.puppeteerPage.waitForNavigation({
-            waitUntil: ['networkidle0'],
-            timeout: networkLoadTimeoutInMilleSec,
-        });
-
-        let response;
+        // separate page load and networkidle0 events to bypass network activity error
+        const gotoUrlPromise = this.puppeteerPage.goto(url, { waitUntil: 'load', timeout: this.pageNavigationTimeoutMsecs });
```
Comment (AhmedAbdoOrtiga) [inline packages/cli/src/scanner/page.ts:47]: so if we wait in here, why do we still need 'waitForPageToCompleteRendering'?

File: packages/cli/src/web-driver/web-driver.ts (hunk 1 above)
Comment (lamaks) [reply packages/cli/src/web-driver/web-driver.ts:16]: It works as it is null by default, we do not pass any value when run app.

File: packages/cli/src/scanner/page.ts (hunk 3 above)
Comment (lamaks) [reply packages/cli/src/scanner/page.ts:123]: yes, this should be added to crawler as well

File: packages/cli/src/scanner/page.ts (hunk 4 above)
Comment (lamaks) [reply packages/cli/src/scanner/page.ts:47]: some pages will have active javascript after load event that affect page rendering and as result color contrast detection. This results in false positive accessibility issues from axe.

**Pull request merged**